sdk/identity/msal.yml

```diff
@@ -0,0 +1,12 @@
+
+trigger: none
+pr: none
+
+extends:
+  template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
+  parameters:
+    ServiceDirectory: identity
+    InjectedPackages: 'git+https://github.com/AzureAD/microsoft-authentication-library-for-python@dev'
```

Comment:
Do we really want to make this it's own pipeline as opposed to a matrix entry in the existing pipeline?

Reply:
We didn't want to run this step as part of every PR or CI and run just once as scheduled. Second point was to avoid msal breakage being a roadblock for any change in azure-identity side. Adding @chlowell  for any additional comments.

Reply:
So it is similar in nature to https://github.com/Azure/azure-sdk-for-python/blob/master/eng/pipelines/templates/jobs/tests-nightly-python.yml where we are doing some validation on a schedule but not really blocking anything on it. If so do we have a plan for who will monitor the failures?

Reply:
Right, we want to catch a breaking change (from the perspective of our expectations) in MSAL's dev branch before it ships. There's no need to run this on our PRs, or more often than daily.